### PR TITLE
fix(ext/web): fix `AbortSignal.timeout()` leak

### LINKED ISSUE #20663: AbortSignal.timeout() causes Deno.test() to report leaking resources?
I'm using `AbortSignal.timeout()` in fetch requests init object, when testing my concurrent request plugin i discovered that my tests fail, not because they don't work, but because deno is saying:

```
error: AssertionError: Test case is leaking async ops.

 - 1 async operation to sleep for a duration was started in this test, but never completed. This is often caused by not cancelling a `setTimeout` or `setInterval` call.
```

I mean it's not saying it explicitely, but my strong suspicion is that the automatic timeout() on the AbortSignal is what causes this...the problem is, how would i fix that? AFAIK there is no way to stop that timeout like we can with clearTimeout on regular setTimeouts, or didn't I dig deep enough into MDN?

EDIT: This is what is reported additionally when --trace-ops is added:

```
The operation was started here:
    at handleOpCallTracing (internal:core/01_core.js:237:42)
    at Object.op_sleep (eval at genAsyncOp (internal:core/01_core.js:181:14), <anonymous>:15:21)
    at Object.opAsync (internal:core/01_core.js:249:12)
    at runAfterTimeout (internal:ext/web/02_timers.js:219:31)
    at initializeTimer (internal:ext/web/02_timers.js:185:5)
    at setTimeout (internal:ext/web/02_timers.js:319:12)
    at Function.timeout (internal:ext/web/03_abort_signal.js:50:25)
```
so it looks like the suspicion was correct... well, now what can we do to fix this?

as far as a workaround disabling `sanitizeResources` in the tests probably will do, but i like that sanitizer and would prefer keeping it active (for other leaky stuff), so if there is a fix or other workaround (other than waiting for the entire timeout at the end of the test of course, lol) it would be good to know.

EDIT 2: Turns out i need to set both of those to false to make the actual tests pass without being the sanitizers the reason they fail:

```
	sanitizeResources: false,
	sanitizeOps: false,
```

### PR DESCRIPTION
**Pull request by tdb-alcorn**
Fixes #20663.

### CODE REVIEW

**Comment by 0f-0b** [comment]
#20663 is saying that a test like this should pass.

```js
Deno.test({
  name: "regression for #20663",
  fn: () => {
    AbortSignal.timeout(2000);
  },
});
```

File: ext/web/03_abort_signal.js [hunk 1]
```diff
@@ -120,6 +120,7 @@ class AbortSignal extends EventTarget {
     const signal = new AbortSignal(illegalConstructorKey);
     signal[timerId] = setTimeout(
```
Comment (mmastrac) [inline ext/web/03_abort_signal.js:121]: Note that Deno.core offers a `queueSystemTimer` to work around this sort of issue. The changes here will still be leaking a timer, as the timer is only ever cleared when the timer has fired (which should have already cleared the timer).

File: tests/unit/timers_test.ts [hunk 2]
```diff
@@ -767,3 +767,15 @@ Deno.test({
     assert(result >= 1000);
   },
 });
+
+// Regression test for https://github.com/denoland/deno/issues/20663
+Deno.test({
+  name: "regression for #20663",
+  fn: async () => {
+    const signal = AbortSignal.timeout(2000);
+    const t = setTimeout(() => {
```
Comment (mmastrac) [inline tests/unit/timers_test.ts:776]: To properly test this fix, I believe you should create an AbortSignal.timeout and let it just leak out of the test.

**Comment by iuioiua** [comment]
Thanks for the `deno_core` fix, @littledivy! Are you able to review this once more?

**Review by iuioiua — APPROVED** [review]

File: Cargo.toml [hunk 3]
```diff
@@ -368,3 +368,6 @@ opt-level = 3
 opt-level = 3
 [profile.release.package.base64-simd]
 opt-level = 3
+
+[patch.crates-io]
+deno_core = { git = "https://github.com/denoland/deno_core" }
```
Comment (iuioiua) [inline Cargo.toml:373]: I missed this before merging. My mistake.

**Pull request merged**